### PR TITLE
EVG-6600 option to filter commit queue patches

### PR DESCRIPTION
**Pull request by ablack12**
This is in-line with the Newer/Older buttons, and modifies the query rather than the current patch results (John has verified that this is the least awkward approach).

image

### CODE REVIEW

**Comment by ybrill** [comment]
It's probably worth pointing out that this is the same approach as in the new patch page: https://github.com/evergreen-ci/spruce/pull/23 (at least last I checked).

File: model/patch/db.go [hunk 1]
```diff
@@ -83,14 +83,25 @@ func ById(id mgobson.ObjectId) db.Q {
...
 }
 
+var commitQueueFilter = bson.M{"$ne": evergreen.CommitQueueAlias}
+
 // ByProject produces a query that returns projects with the given identifier.
-func ByProject(project string) db.Q {
-	return db.Query(bson.M{ProjectKey: project})
+func ByProject(project string, filterCommitQueue bool) db.Q {
+	q := bson.M{ProjectKey: project}
+	if filterCommitQueue {
```
Comment (ybrill) [inline model/patch/db.go]: This seems unconnected to the ByProject and ByUser queries.
Maybe we could set the Alias predicate on the return value in the [callers](https://github.com/evergreen-ci/evergreen/blob/3af9096f5f3bc045f777ddfc71e28efe85a86399/service/timeline.go#L118)?

On the other hand, it seems like this is the only place we use these functions for the time being.

File: service/templates/patches.html [hunk 2]
```diff
@@ -28,22 +28,25 @@ <h1>
...
-        </a>
-      </span>
-      <span>
-        <a  class="btn btn-default"
-            style="cursor: pointer"
-            ng-click="nextPage()">
-          Older
-        </a>
-      </span>
+    <div class="form-inline" ng-if="patches.length > 0">
```
Comment (ybrill) [inline service/templates/patches.html]: in the (albeit unlikely) scenario where filtering the patches results in 0 patches will there be a way to remove the filter?

File: service/templates/patches.html [hunk 3]
```diff
@@ -28,22 +28,25 @@ <h1>
...
+      <div class="btn-group btn-group-sm header-pagination">
+        <input type="checkbox" id="queue" ng-model="filterCommitQueue" ng-change="toggleFilter()">
```
Comment (ybrill) [inline service/templates/patches.html]: is there a way to make this setting persist on subsequent page loads?

File: model/patch/db.go (hunk 1 above)
Comment (ablack12) [reply model/patch/db.go]: I can rename the functions to be more clear

File: service/templates/patches.html (hunk 2 above)
Comment (ablack12) [reply service/templates/patches.html]: Ah weird edge case. I'll move the ng-if down past the checkbox.

Comment (ablack12) [reply service/templates/patches.html]: Actually I noticed this happened when I scrolled to the end of my patches in my local evergreen too, so I'll just get rid of it overall.

File: service/templates/patches.html (hunk 3 above)
Comment (ablack12) [reply service/templates/patches.html]: Done!

**Comment by ablack12** [comment]
@ybrill this can be reviewed again

File: service/templates/patches.html [hunk 4]
```diff
@@ -28,22 +28,25 @@ <h1>
...
+        <input type="checkbox" id="queue" ng-model="filterCommitQueue">
+        <label for="queue" style="font-weight:normal;margin-right: 10px;">Filter Commit Queue</label>
+        <span ng-if="currentPage > 0">
+          <a  class="btn btn-default"
+              style="cursor: pointer"
+              ng-click="previousPage()">
+            Newer
+          </a>
+        </span>
+        <span>
```
Comment (ybrill) [inline service/templates/patches.html]: maybe it makes sense to make this span conditional on whether there is a next page, or at least that the number of patches on the current page is equal to the limit (so it's possible there is a next page)?

Comment (ablack12) [reply service/templates/patches.html]: I'm not sure these buttons are pervasive enough that it's a problem to have them--to know whether there is a next page would take another API call, and the limit is configured on the go side. https://github.com/evergreen-ci/evergreen/blob/master/service/timeline.go#L113

Comment (ablack12) [reply service/templates/patches.html]: I'll add an if statement to have the older button disappear if there are no patches.

**Pull request merged**